### ISSUE TITLE
Trino - Found unparsable section with Row datatype definition at CAST

### ISSUE THREAD

**Issue by bonisb** [sqlfluff/sqlfluff#6084]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When renaming a ROW object's fields by casting the ROW with types and names, the parser fails as the parser does not recognise the ROW structure.

```sql
SELECT
    name,
    CAST(ROW(price, store) AS ROW(price REAL, store VARCHAR)) AS data_row
FROM customers
```

SQL fluff fails with a parser error:
```
L:   7 | P:  14 |  PRS | Line 7, Position 14: Found unparsable section:
                       | 'REAL, store VARCHAR'
```

### Expected Behaviour

sqlfluff should support constructing ROW datatypes at casting columns with Trino dialect. (Like Bigquery does with STRUCT)

### Observed Behaviour

sqlfluff recognises the ROW keyword, but does not recognise the ROW datatype expression.

### How to reproduce

```sql
SELECT
    name,
    CAST(ROW(price, store) AS ROW(price REAL, store VARCHAR)) AS data_row
FROM customers
```

```
$ sqlfluff lint --dialect trino test_query.sql
```

### Dialect

Trino

### Version

3.1.0

### Configuration

default

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
